src/main/daml/Daml/Finance/Instrument/Bond/Lifecycle/Rule.daml

```diff
@@ -0,0 +1,131 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Instrument.Bond.Lifecycle.Rule
+  ( Rule(..)
+  ) where
+
+import DA.Foldable (forA_)
+import DA.Optional (fromSome)
+import DA.Set (fromList)
+import DA.Text (sha256)
+import Daml.Finance.Claims.Util (isZero')
+import Daml.Finance.Claims.Util.Lifecycle (electionEvent, lifecycle, lifecycleClaims, splitPending)
+import Daml.Finance.Instrument.Bond.Callable.Instrument (Instrument(..))
+import Daml.Finance.Interface.Claims.Dynamic.Instrument qualified as DynamicInstrument (I, GetView(..))
+import Daml.Finance.Interface.Claims.Types (ElectionData(..))
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as BaseInstrument (R, createReference, exerciseInterfaceByKey, getKey)
+import Daml.Finance.Interface.Lifecycle.Election qualified as Election (ApplyElection(..), Exercisable(..), ExercisableView(..), getElectionTime)
+import Daml.Finance.Interface.Lifecycle.Rule.Lifecycle qualified as Lifecycle (I, Evolve(..), View(..))
+import Daml.Finance.Interface.Types.Common.Types (Id(..), InstrumentKey, Parties, PartiesMap)
+import Daml.Finance.Interface.Util.Common (fetchInterfaceByKey)
+import Daml.Finance.Interface.Util.Disclosure qualified as Disclosure (I, View(..), flattenObservers)
+import Daml.Finance.Lifecycle.ElectionEffect (ElectionEffect(..))
+import Daml.Finance.Util.Disclosure (addObserversImpl, removeObserversImpl, setObserversImpl)
+import Prelude hiding (exercise, key)
+
+-- | Rule to process an election event for options.
+template Rule
```

Comment:
No chance to re-use the rule for dynamic claim instruments?

Reply:
Yes, I have added an issue for this: https://github.com/digital-asset/daml-finance/issues/739
I would prefer to do those kind of changes in a separate PR, since it impacts other instruments as well.